File: 1904-second-largest-digit-in-a-string/second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
        int first =-1;
        int second = -1;

        for(char ch:s.toCharArray()){
            if(Character.isDigit(ch)){
                int val = ch-'0';
                if (val>first){
                    second=first;
                    first = val;
                }
                else if (val>=second && val!=first){
                    second = val;
                }
            }
        }
        return second;
    }
}

File: 1904-second-largest-digit-in-a-string/second-largest-digit-in-a-string.py
class Solution:
    def secondHighest(self, s: str) -> int:
        first,second=-1,-1
        for ch in s:
            if ch.isdigit():
                c=int(ch)
                if c>first:
                    second,first=first,c
                if second <c<first:
                    second =c
        return second